pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -735,34 +748,65 @@ func (r *ReferenceAdapter) AddOrUpdateBindings(serviceAccountEmail string, polic
 func (r *ReferenceAdapter) SetIAMPolicy(serviceAccountEmail string, policies []string, memberType gcputil.IamMemberType) error {
 	// Checking if policy needs to be updated
 	var retry int
+	addorUpdateResponse, err := r.AddOrUpdateBindings(serviceAccountEmail, policies, memberType)
+	if err != nil {
+		return err
+	}
+	if !addorUpdateResponse.modified {
+		return nil
+	}
+	setIamPolicyRequest := &cloudresourcemanager.SetIamPolicyRequest{
+		Policy: addorUpdateResponse.policy,
+	}
+
 	for {
```

Comment:
This logic can be a separate function be very careful with the deletion logic. If we can put the deletion logic function in the utils package.